src/main/java/task/Task.java

```diff
@@ -0,0 +1,98 @@
+package task;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a task in the Duke application.
+ */
+public class Task {
+    protected String symbol;
+    protected String description;
+    protected boolean isDone;
+    protected LocalDate by;
+    protected LocalDate start;
+    protected LocalDate end;
+
+    /**
+     * Constructs a Task with the specified symbol and description.
+     *
+     * @param symbol      the symbol representing the type of the task
+     * @param description the description of the task
+     */
+    public Task(String symbol, String description) {
+        this.symbol = symbol;
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a Task with the specified symbol, description, and deadline date.
+     *
+     * @param symbol      the symbol representing the type of the task
+     * @param description the description of the task
+     * @param by          the deadline date for tasks with a specified deadline
+     */
+    public Task(String symbol, String description, LocalDate by) {
+        this.symbol = symbol;
+        this.description = description;
+        this.isDone = false;
+        this.by = by;
+    }
+
+    /**
+     * Constructs a Task with the specified symbol, description, start date, and end date.
+     *
+     * @param symbol      the symbol representing the type of the task
+     * @param description the description of the task
+     * @param start       the start date for events with a specified duration
+     * @param end         the end date for events with a specified duration
+     */
```

Comment:
```suggestion
    /**
     * Constructs a Task with the specified symbol, description, start date, and end date.
     *
     * @param symbol      The symbol representing the type of the task.
     * @param description The description of the task.
     * @param start       The start date for events with a specified duration.
     * @param end         The end date for events with a specified duration.
     */
```
I really like how you start the description for the params at the same starting point as it looks very neat!
Perhaps it would be better if you start each description with a capital letter and ends of with a full stop?